eo-maven-plugin/src/test/java/org/eolang/maven/MarkMojoTest.java

```diff
@@ -65,16 +71,45 @@ void updatesVersionIfItExists(@TempDir final Path temp) throws IOException {
         maven.execute(MarkMojo.class);
         MatcherAssert.assertThat(
             foreign.all().iterator().next().version(),
-            Matchers.equalTo("0.1.8")
+            Matchers.equalTo(MarkMojoTest.VERSION)
         );
         MatcherAssert.assertThat(
             foreign.size(),
             Matchers.equalTo(1)
         );
     }
 
+    @Test
+    void extendsTojosWithVersionedOne(@TempDir final Path temp) throws IOException {
+        MarkMojoTest.source(temp);
+        final ForeignTojos tojos = new FakeMaven(temp)
+            .with("withVersions", true)
+            .execute(MarkMojo.class)
+            .externalTojos();
+        final ObjectName object = new OnVersioned("foo.bar", "6a70071");
+        MatcherAssert.assertThat(
+            String.format(
+                "Tojos should have contained versioned object %s after extending, but they didn't",
+                object
+            ),
+            tojos.contains(object),
+            Matchers.is(true)
+        );
+    }
+
+    /**
+     * Save foo/bar.eo file to target/hello/-/0.28.0/EO-SOURCES.
```

Comment:
@maxonfjvipon It is dangerous to duplicate logic in comments, as the logic might change without corresponding updates to the comments. Moreover, why should we state the same thing twice? The code already expresses what is going on clearly.